### PR TITLE
Add m3o.com to PaaS section

### PR DESCRIPTION
**Pull request by asim**
Adds m3o.com to the PaaS section. M3O is based on the open source project Micro but provides a fully managed and hosted offering including a free tier for devs known as m3o.dev.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -643,6 +643,7 @@ Table of Contents
   * [K8Spin](https://k8spin.cloud) - Managed Kubernetes namespace service designed for developers. Free developer accounts come with with 100 mCore of CPU, 128MB of RAM.
   * [ZARVIS](https://zarvis.ai) - Free managed Kubernetes namespace for open source Github project. Free 1GB memory and 1 vCPU.
   * [mendix.com](https://www.mendix.com/) — Rapid Application Development for Enterprises, unlimited number of free sandbox environments supporting 10 users, 100 MB of files and 100 MB database storage each
+  * [m3o.com](https://m3o.com) - A cloud platform for API services development. M3O is a fully managed Micro as a Service offering focusing on Go microservices development in the Cloud. Free tier provides enough to run 5 services and collaborate with others.
```
Comment (ripienaar) [inline README.md:646]: Is the signup processes documented there all thats needed for the free tier or MUST you first do `To get access please talk to #m3o-platform on slack.`?

Comment (asim) [reply README.md:646]: Signup process is documented on the landing page as `micro signup`. Its still invite only by way of talking to us.

Comment (ripienaar) [reply README.md:646]: OK, unfortunately in the past we've not added invite only/human based signup free services here, so afraid I'll have to do the same here.  Has to be low touch

Comment (asim) [reply README.md:646]: OK. Please leave this here for now. We'll open up the dev tier without invite requirements on October 20th. So I'll ping again when its done.

Comment (ripienaar) [reply README.md:646]: ok sweet

Comment (asim) [reply README.md:646]: Hey @ripienaar. The Dev tier went live today. Developers will see instructions on the website and related docs for using the free environment by simply switching using `micro env set dev`. Can you please merge this PR at your earliest convenience. Thanks.

Comment (ripienaar) [reply README.md:646]: https://m3o.dev/concepts/platform#dev-environment there? still asks for slack or do you mean you're in process of rolling that out?

Comment (asim) [reply README.md:646]: Fixed. Its just a wording issue. The sign-up is totally open.

Comment (ripienaar) [reply README.md:646]: good stuff, thanks for sticking with it :)

Comment (asim) [reply README.md:646]: Thank you for merging!

**Pull request merged**